delta/plugins/composite-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/indexing/CompositeViewsCoordinator.scala

```diff
@@ -62,39 +62,22 @@ final class CompositeViewsCoordinator(
                             } yield ()
                           )
           } yield ()
-        case d: DeprecatedViewDef  => cleanupCurrent(d.ref)
+        case d: DeprecatedViewDef  => cleanupCurrent(cache, d, destroy)
       }
     }
   }
 
-  /**
-    * If a previous version of the view is running: * Stop it * Delete the associated namespaces and indices * Delete
-    * existing progress
-    */
-  private def cleanupCurrent(ref: ViewRef): Task[Unit] =
-    cache.get(ref).flatMap {
-      case Some(v) =>
-        supervisor
-          .destroy(
-            v.projection,
-            for {
-              _ <-
-                Task.delay(
-                  logger.info(
-                    s"View '${ref.project}/${ref.viewId}' has been updated or deprecated, cleaning up the current one."
-                  )
-                )
-              _ <- buildSpaces(v).destroy
-              _ <- compositeProjections.deleteAll(v.ref, v.rev)
-              _ <- cache.remove(v.ref)
-            } yield ()
-          )
-          .void
-      case None    =>
-        Task.delay(
-          logger.debug(s"View '${ref.project}/${ref.viewId}' is not referenced yet, cleaning is aborted.")
-        )
-    }
+  private def destroy(active: ActiveViewDef) =
```

Comment:
If we don't have any abstraction between coordinators, it would be nice to ultimately have the destroy function in ES and BG coordinators as well. The bug was easy to spot when comparing the coordinators as they had the same structure.

Reply:
We can see if it is worth it even if if an abstraction for 3 cases may be overkill and might be not bring a real gain in the end (the previous implementation in 1.7 had one and it was not a real win).
We must also do this refactor for the construct query before looking for it